news/bugfix-3892.md

```diff
@@ -0,0 +1,5 @@
+`file()` source: fixed invalid buffer handling when `encoding()` is used
+
+A bug has been fixed that - under rare circumstances - could cause message
+duplication, partial message loss, even crashes when non-fixed length or less
```

Comment:
do we have security implications here?

Reply:
I'm still investigating the crash part of it, but AFAIK this should have no security implications, because we always stay within the buffer's boundaries, we just mess up the positions within those boundaries.

Reply:
The crash I reproduced was a bug in my tests. I'm removing the crash part from the news entry.